src/path.c

```diff
@@ -249,6 +249,40 @@ ly_path_check_predicate(const struct ly_ctx *ctx, const struct lysc_node *cur_no
     return LY_EVALID;
 }
 
+/**
+ * @brief Skips already parsed XPath FunctionName the corresponding schema node
+ *
+ * @param[in] ctx libyang context.
+ * @param[in] exp Parsed path.
+ * @param[in,out] tok_idx Index in @p exp, is adjusted.
+ * @return LY_ERR value.
+ */
+static LY_ERR
+ly_path_skip_function(const struct ly_ctx *ctx, const struct lyxp_expr *exp, uint32_t *tok_idx)
+{
+    LY_ERR ret = LY_SUCCESS;
+    /* mandatory FunctionName */
+    LY_CHECK_ERR_GOTO(lyxp_next_token(ctx, exp, tok_idx, LYXP_TOKEN_FUNCNAME), ret = LY_EVALID, error);
+
+    /* mandatory '(' */
+    LY_CHECK_ERR_GOTO(lyxp_next_token(ctx, exp, tok_idx, LYXP_TOKEN_PAR1), ret = LY_EVALID, error);
+
+    /* mandatory 'Expr' */
+    while (lyxp_check_token(NULL, exp, *tok_idx, LYXP_TOKEN_PAR2) && lyxp_check_token(NULL, exp, *tok_idx, LYXP_TOKEN_FUNCNAME)){
+         (*tok_idx)++;
+    }
+
+    /* embedded function inside function ')' */
+    if (!lyxp_check_token(NULL, exp, *tok_idx, LYXP_TOKEN_FUNCNAME)) {
```

Comment:
A more robust skip would be preferred such as counting the parentheses, for example.

Reply:
This depends how much you want to restrict this feature. Either you want to allow just "deref(relative-path-arg)" or generic XPath. Current check is validating just generic XPath, by relaying on generic xpath parser and validating just what is needed to get to the last part of relative path section. I am not sure how to make it better as there can be many complex scenarios of nested structures etc...

Do you have any exact proposal?

Reply:
Right, the argument is supposed to again be a leafref relative/absolute path, which is already being parsed so there are all the required functions for parsing it. And yes, you should also check just for `deref()` as only that function should be allowed.

Reply:
Ok, done